typescript/index.d.ts

```diff
@@ -29,7 +29,11 @@
  export type RecoilRootProps = {
   initializeState?: (mutableSnapshot: MutableSnapshot) => void,
   override?: true,
- } | {override: false};
+  children?: React.ReactNode | undefined,
```

Comment:
Let's use this opportunity to make it required and align with the Flow type https://github.com/facebookexperimental/Recoil/blob/main/packages/recoil/core/Recoil_RecoilRoot.js#L66

```suggestion
  children: React.ReactNode,
```

(also I think `undefined` is included in `React.ReactNode`)